@yongtang This test failure seems to be related to this PR:

======================================================================
ERROR: testShapeInference (__main__.ResizeImageWithCropOrPadTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_test.py", line 2305, in testShapeInference
    self._assertShapeInference([None, 69, 3], 55, 66, [55, 66, 3])
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_test.py", line 2155, in _assertShapeInference
    y = image_ops.resize_image_with_crop_or_pad(image, height, width)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_impl.py", line 659, in resize_image_with_crop_or_pad
    min_(target_width, width))
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/ops/image_ops_impl.py", line 565, in crop_to_bounding_box
    cropped.set_shape(cropped_shape)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/ops.py", line 419, in set_shape
    self._shape = self._shape.merge_with(shape)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local_linux-opt/bin/tensorflow/python/image_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/tensor_shape.py", line 582, in merge_with
    raise ValueError("Shapes %s and %s are not compatible" % (self, other))
ValueError: Shapes (?, ?, 65, ?) and (1, ?, 66, 3) are not compatible

----------------------------------------------------------------------
Ran 27 tests in 8.282s

FAILED (errors=1)
================================================================================

--- a/tensorflow/core/ops/array_ops.cc
+++ b/tensorflow/core/ops/array_ops.cc
@@ -2293,24 +2293,6 @@ Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
 
   return Status::OK();
 }
-
-// This SliceHelper processes the output shape of the `slice`
-// when the tensor of `sizes` is not available (null).
-// In this situation, we could still obtain partial information
-// from `MakeShapeFromShapeTensor` as a best effort operation.
-Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
-                   ShapeHandle sizes_value,
-                   std::vector<DimensionHandle>* dims) {
-  for (int i = 0; i < c->Rank(sizes_value); ++i) {
-    DimensionHandle dim = c->Dim(sizes_value, i);
-    DimensionHandle begin = c->Dim(begin_value, i);
-    DimensionHandle result = c->UnknownDim();
-    TF_RETURN_IF_ERROR(c->Subtract(dim, begin, &result));
-    dims->emplace_back(result);
-  }
-
-  return Status::OK();
-}
 }  // namespace
 
 // --------------------------------------------------------------------------
@@ -2379,7 +2361,9 @@ REGISTER_OP("Slice")
           TF_RETURN_IF_ERROR(
               c->WithRank(begin_value, c->Rank(sizes_value), &begin_value));
           std::vector<DimensionHandle> dims;
-          TF_RETURN_IF_ERROR(SliceHelper(c, begin_value, sizes_value, &dims));
+          for (int i = 0; i < c->Rank(sizes_value); ++i) {
+            dims.emplace_back(c->Dim(sizes_value, i));
+          }
           c->set_output(0, c->MakeShape(dims));
           return Status::OK();
         }

